go/protocol/keybase1/extras.go

```diff
@@ -2625,3 +2611,59 @@ func (se *SelectorEntry) UnmarshalJSON(b []byte) error {
 	}
 	return fmt.Errorf("invalid selector (not recognized)")
 }
+
+type UPKLiteInterface interface {
+	GetUID() UID
+	GetDeviceKeys() map[KID]PublicKeyV2NaCl
+	ToUserVersion() UserVersion
+}
+
+type UPKLiteAllIncarnationsInterface interface {
+	GetCurrent() UPKLiteInterface
+	GetPastIncarnations() []UPKLiteInterface
+	GetSeqnoLinkIDs() map[Seqno]LinkID
+}
+
+func (u UpkLiteV1) GetDeviceKeys() map[KID]PublicKeyV2NaCl {
+	return u.DeviceKeys
+}
+
+func (u UpkLiteV1AllIncarnations) GetCurrent() UPKLiteInterface {
+	return u.Current
+}
+
+func (u UpkLiteV1AllIncarnations) GetPastIncarnations() []UPKLiteInterface {
+	ret := make([]UPKLiteInterface, len(u.PastIncarnations))
+	for idx, incarnation := range u.PastIncarnations {
```

Comment:
also pretty annoying, but in Go if Foo implements FooInterface, []Foo is still not a []FooInterface